client/src/rpc_client.rs

```diff
@@ -128,12 +131,42 @@ impl RpcClient {
         self.send_transaction_with_config(transaction, RpcSendTransactionConfig::default())
     }
 
+    fn default_cluster_transaction_encoding(&self) -> Result<UiTransactionEncoding, RpcError> {
+        let default_cluster_transaction_encoding =
+            self.default_cluster_transaction_encoding.read().unwrap();
+        let encoding = if let Some(encoding) = *default_cluster_transaction_encoding {
+            encoding
```

Comment:
nit: I think an early return here would have a positive effect on overall readability 
```suggestion
            return Ok(encoding)l
```

Reply:
Specifically to communicate the quick path?  'cause otherwise I'll just `Ok(encoding)` the `else`-block too, drop the outer assignment and omit the `return` here

Reply:
Did that :point_up: .  We can add the `return` in later for better self-doc if need be